py/desiutil/plots.py

```diff
@@ -206,55 +272,32 @@ def prepare_data(data, mask=None, clip_lo=None, clip_hi=None,
 
     These attributes can then be used by plotting routines to fix the input
     range used for colormapping, independently of the actual range of data.
-
-    Parameters
-    ----------
-    data : array or masked array
-        Array of data values to assign colors for.
-    mask : array of bool or None
-        Array of bools with same shape as data, where True values indicate
-        values that should be ignored when assigning colors.  When None, the
-        mask of a masked array will be used or all values of an unmasked
-        array will be used.
-    clip_lo : float or str
-        Data values below clip_lo will be clipped to the minimum color. If
-        clip_lo is a string, it should end with "%" and specify a percentile
-        of un-masked data to clip below.
-    clip_hi : float or str
-        Data values above clip_hi will be clipped to the maximum color. If
-        clip_hi is a string, it should end with "%" and specify a percentile
-        of un-masked data to clip above.
-    save_limits : bool
-        Save the calculated lo/hi clip values as attributes vmin, vmax of
-        the returned masked array.  Use this flag to indicate that plotting
-        functions should use these vmin, vmax values when mapping the
-        returned data to colors.
-
-    Returns
-    -------
-    masked array
-        Masked numpy array with the same shape as the input data, with any
-        input mask applied (or copied from an input masked array) and values
-        clipped to [clip_lo, clip_hi].
     """
     data = np.asanyarray(data)
     if mask is None:
         try:
             # Use the mask associated with a MaskedArray.
-            mask = data.mask
+            cmask = data.mask
             # If no clipping is requested, pass the input through.
             if clip_lo is None and clip_hi is None:
                 return data
         except AttributeError:
             # Nothing is masked by default.
-            mask = np.zeros_like(data, dtype=bool)
+            cmask = np.zeros_like(data, dtype=bool)
     else:
-        mask = np.asarray(mask)
-        if mask.shape != data.shape:
+        #
+        # Make every effort to ensure that modifying the mask of the output
+        # does not modify the input mask.
```

Comment:
Note: `np.asarray(x)` will not make a copy if `x` is already an array, while `np.array(x)` will make a copy.  Both will promote lists to arrays if needed (thus giving more input flexibility than `x.copy()`).  i.e. I think the code below could have been implemented as simply `cmask = np.array(mask)` without needing try/except.